File: src/main/java/com/lcaohoanq/nocket/domain/reaction/ReactionRepository.java
package com.lcaohoanq.nocket.domain.reaction;

import com.lcaohoanq.nocket.enums.EReaction;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ReactionRepository extends JpaRepository<Reaction, UUID> {

    Optional<Reaction> findByReaction(EReaction reactionType);


}

File: src/main/java/com/lcaohoanq/nocket/domain/reaction/ReactionRepository.kt
package com.lcaohoanq.nocket.domain.reaction

import com.lcaohoanq.nocket.enums.EReaction
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ReactionRepository : JpaRepository<Reaction, UUID> {
    fun findByReaction(reactionType: EReaction): Optional<Reaction>
}
